api/advert_api.go

```diff
@@ -27,6 +27,7 @@ type AdvertAPI struct {
 func NewAdvertAPI(r *mux.Router, log *logrus.Entry, app application.Application, middleware *MiddlewareProvider, sessionStore *sessions.CookieStore, cfg *common.Config) *AdvertAPI {
 	advertApi := AdvertAPI{router: r, app: app, log: log, sessionStore: sessionStore, cfg: cfg}
 	r.HandleFunc("/api/adverts", middleware.AuthMiddleware(advertApi.AddAdvert, log)).Methods("POST")
+	r.HandleFunc("/api/adverts", middleware.AuthMiddleware(advertApi.AdvertsList, log)).Methods("GET")
```

Comment:
we could create handler that accepts controller func and applies middlewares from slice maybe? e.g Handle(advertApi.AdvertsList, []Handler{middleware.AuthMiddleware})

Reply:
Could do, but at this point there's only 1 middleware at handler level (and more middlewares at router level)/

btw. this endpoint shouldn't have AuthMiddleware as it's just list of adverts and everyone should be able to access it